server/webhook.go

```diff
@@ -219,10 +220,20 @@ func (p *Plugin) permissionToProject(ctx context.Context, userID, namespace, pro
 	})
 	if result == nil || err != nil {
 		if err != nil {
-			p.client.Log.Warn("can't get project in webhook", "err", err.Error(), "project", namespace+"/"+project)
+			p.client.Log.Warn("Can't get project in webhook", "err", err.Error(), "project", namespace+"/"+project)
 		}
 		return false
 	}
+
+	if result.Permissions.ProjectAccess == nil {
+		return false
+	}
+
+	// Check for guest level permissions
+	if result.Permissions.ProjectAccess != nil && result.Permissions.ProjectAccess.AccessLevel == gitlabLib.GuestPermissions {
```

Comment:
We doin't need the `nil` check here because we check above

```suggestion
	if result.Permissions.ProjectAccess.AccessLevel == gitlabLib.GuestPermissions {
```